src/frontend/sharedTypes.ts

```diff
@@ -0,0 +1,41 @@
+// TS port of /src/frontend/types.js
+import { ImageStyle, StyleProp, TextStyle, ViewStyle } from "react-native";
+import { NavigationRoute } from "react-navigation";
+import {
+  NavigationStackProp,
+  NavigationStackScreenComponent,
+} from "react-navigation-stack";
+
+export type ViewStyleProp = StyleProp<ViewStyle>;
+export type TextStyleProp = StyleProp<TextStyle>;
+export type ImageStyleProps = StyleProp<ImageStyle>;
+
+export type MapStyle = {
+  id: string;
+  name: string;
+  bounds: number[];
+  minzoom: number;
+  maxzoom: number;
+};
+
+export type IconSize = "small" | "medium" | "large";
+export type ImageSize = "thumbnail" | "preview" | "original";
+
+export type Status = "idle" | "loading" | "error" | "success" | void;
+
+type NavigationParams = {
+  observationId?: string;
+  question: number;
+  photoIndex?: number;
+  editing?: boolean;
+  handleSavePress?: () => void;
+};
+
+export type NavigationProp = NavigationStackProp<
+  NavigationRoute,
+  NavigationParams
+>;
+
+export type StackScreenComponent = NavigationStackScreenComponent<
+  NavigationParams
+>;
```

Comment:
It does seem like all the shared info is shared through our context providers, so I would say for simplicity we can just name the navigation params directly in the component. But that is a personal preference, and I think we can do it both ways unless you have any strong opposition to that!

Reply:
yeah it's probably sufficient to do define this per component. this just based on whatever was defined in `types.js`

Reply:
I _think_ these navigation params hold data that is not in context providers - this is the part of the state that is held in the navigation. However it's been a while since I looked at this. It is in prep for deep-linking support, e.g. a link that opens a screen in the app directly for a particular observation or photo, should we need that in the future.

Reply:
i won't make big adjustments to this in that case. can always follow up if needed